### PR TITLE
Add linear testsetup 

### PR DESCRIPTION
**Pull request by sopje**
To easily actuate the linear test setup, a different launch file for the linear test setup is created. Also, the naming testjoint is not being used anymore, instead it is rotational or linear test joint. Also, a xacro including the correct limits of the linear test joint has been made. This pull request is part of this pull request: https://github.com/project-march/hardware-interface/pull/114

### CODE REVIEW

File: march_description/urdf/test-joint-lin.xacro [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    </link>
+
+
+    <!-- ROS Control plugin for Gazebo -->
+    <transmission name="test_joint">
+        <type>transmission_interface/SimpleTransmission</type>
+        <joint name="test_joint">
+            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
+        </joint>
+        <actuator name="test_joint_motor">
```
[the remark this answers is not in the document]
Comment (sopje) [reply march_description/urdf/test-joint-lin.xacro]: updated it

File: march_description/urdf/test-joint-lin.xacro [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    <!-- Used for fixing robot to Gazebo 'base_link' -->
+    <link name="world"/>
+
+    <joint name="joint_base_fixed_parent_joint" type="fixed">
+        <parent link="world"/>
+        <child link="joint_base"/>
+        <origin xyz="0 0 ${hover_height}" rpy="0 0 0"/>
+    </joint>
+
+    <joint name="test_joint_lin" type="revolute">
```
[the remark this answers is not in the document]
Comment (sopje) [reply march_description/urdf/test-joint-lin.xacro]: I  changed it to linear_joint

File: march_description/urdf/test-joint-rot.xacro [hunk 3]
```diff
@@ -40,7 +40,7 @@
         <origin xyz="0 0 ${hover_height}" rpy="0 0 0"/>
     </joint>
 
-    <joint name="test_joint" type="revolute">
+    <joint name="test_joint_rot" type="revolute">
```
[the remark this answers is not in the document]
Comment (sopje) [reply march_description/urdf/test-joint-rot.xacro]: it is now rotational_joint

**Comment by sopje** [comment]
> I agree with the idea, but as this is being refactored, maybe you can think of a good name. `test-joint-lin` is still quite vague, what makes something a 'test setup'.
> 
> Also please use `rotational` and `linear` for more clarity.
> 
> image

I still feel like test_joint_rotational & test_joint_linear are proper names, we always refer to these setups with those names. If we would refer to them as linear or rotational joint solely, it could also be a linear or rotational joint in the exoskeleton, but by naming them so, we specifically refer to a linear or rotational joint in a test environment.

File: march_description/urdf/test-joint-linear.xacro [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+            <inertia
+                    ixx="${joint_base_mass / 12.0 * (bar_width*bar_width + bar_length*bar_length)}" ixy="0.0" ixz="0.0"
+                    iyy="${joint_base_mass / 12.0 * (bar_length*bar_length + bar_width*bar_width)}" iyz="0.0"
+                    izz="${joint_base_mass / 12.0 * (bar_width*bar_width + bar_width*bar_width)}"/>
+        </inertial>
+    </link>
+
+
+    <!-- ROS Control plugin for Gazebo -->
+    <transmission name="test_joint">
```
Comment (Ishadijcks) [inline march_description/urdf/test-joint-linear.xacro]: Fix transmission name

File: march_description/urdf/test-joint-rotational.xacro [hunk 5]
```diff
@@ -98,7 +98,7 @@
     <!-- ROS Control plugin for Gazebo -->
     <transmission name="test_joint">
         <type>transmission_interface/SimpleTransmission</type>
-        <joint name="test_joint">
+        <joint name="rotational_joint">
             <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
         </joint>
         <actuator name="test_joint_motor">
```
Comment (Ishadijcks) [inline march_description/urdf/test-joint-rotational.xacro]: Fix actuator name

File: march_description/urdf/test-joint-rotational.xacro [hunk 6]
```diff
@@ -98,7 +98,7 @@
     <!-- ROS Control plugin for Gazebo -->
     <transmission name="test_joint">
```
Comment (Ishadijcks) [inline march_description/urdf/test-joint-rotational.xacro]: Fix trnasmission name

File: march_description/urdf/test-joint-linear.xacro [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+        <type>transmission_interface/SimpleTransmission</type>
+        <joint name="linear_joint">
+            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
+        </joint>
+        <actuator name="test_joint_motor">
```
Comment (Ishadijcks) [inline march_description/urdf/test-joint-linear.xacro]: Fix actuator name

**Comment by TimBuckers** [comment]
Is this going to be merged? Or should this be closed?

**Comment by Olavhaasie** [comment]
@sopje Is this still relevant? If so the launch files also need to be edited to use the URDF files

**Pull request merged**